ingestor/chalicelib/daily_speeds.py

```diff
@@ -102,7 +102,7 @@ def populate_daily_table(start_date: datetime, end_date: datetime, line: str, ro
     current_date = start_date.date()
     delta = timedelta(days=180)
     speed_objects = []
-    while current_date < end_date:
+    while current_date < end_date.date():
```

Comment:
Shouldn't `end_date` already be a datetime object based on the func type hint?

Reply:
It is a datetime object, but current_date is a _date object, so we need to compare _date to _date